### PR TITLE
Refs 3426. Optimization doc

### PR DESCRIPTION
**Pull request by lemunozm**

### CODE REVIEW

File: docs/optimization.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Executable code size
+--------------------
+In order to manage the executable code size, the library can be compiled enabling or disabling several profiles.
+To add or remove a profile from the library, edit the ``client.config`` file.
+More information can be found at: :ref:`micro_xrce_dds_client_label`.
+
+Runtime size
+------------
+The client is static memory free.
+This implies that all memory charge depends of how the stack grows during the execution.
```
Comment (richiware) [inline docs/optimization.rst]: I think it is `depends on`

File: docs/optimization.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+To control the stack, there are some values that can be modified and impact to a large degree:
+
+Stream buffers
+~~~~~~~~~~~~~~
+1. Make sure you defined only the streams that will be used.
```
Comment (richiware) [inline docs/optimization.rst]: It would be useful a link to the section where stream concept is explained.

File: docs/optimization.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+   You can define a maximun `127` best effort streams and `128` reliable streams,
+   but for the majory of purposes, only one streams of either best effort or reliable can be used.
```
Comment (richiware) [inline docs/optimization.rst]: It should be `only one stream`, should'nt?

File: docs/optimization.rst (hunk 1 above)
Comment (lemunozm) [reply docs/optimization.rst]: Done!

File: docs/optimization.rst (hunk 2 above)
Comment (lemunozm) [reply docs/optimization.rst]: I think that would be useful too :smiley:, but we still do not have a dedicated part for it. Comming soon.
Besides, the optimization part is only useful for advance users, and this users must know already the stream concepts

File: docs/optimization.rst (hunk 3 above)
Comment (lemunozm) [reply docs/optimization.rst]: You are right. Done!

**Pull request merged**